Frontend/CO.CDP.OrganisationApp/Pages/Supplier/SupplierEmailAddress.cshtml.cs

```diff
@@ -0,0 +1,81 @@
+using CO.CDP.Organisation.WebApiClient;
+using CO.CDP.OrganisationApp.WebApiClients;
+using Microsoft.AspNetCore.Mvc;
+using System.ComponentModel.DataAnnotations;
+
+namespace CO.CDP.OrganisationApp.Pages.Supplier;
+
+[AuthorisedSession]
+public class SupplierEmailAddressModel(
+    ISession session,
+    IOrganisationClient organisationClient) : LoggedInUserAwareModel
+{
+    public override ISession SessionContext => session;
+
+    [BindProperty]
+    [Required(ErrorMessage = "Enter your email address")]
+    [EmailAddress(ErrorMessage = "Enter an email address in the correct format, like name@example.com")]
+    public string? EmailAddress { get; set; }
+
+    [BindProperty(SupportsGet = true)]
+    public Guid Id { get; set; }
+
+    public async Task<IActionResult> OnGet(Guid id)
+    {
+        try
+        {
+            var getOrganisationTask = organisationClient.GetOrganisationAsync(id);
+            var getSupplierInfoTask = organisationClient.GetOrganisationSupplierInformationAsync(id);
+
+            await Task.WhenAll(getOrganisationTask, getSupplierInfoTask);
```

Comment:
Just a note for future PRs: we repeat these 3 lines of code quite a lot (or possibly more if we include checking the result below).

Looks like there's a collaborator object wanting to get out here.